dotnet/src/Connectors/VectorData.Abstractions/VectorStoreGenericDataModel.cs

```diff
@@ -8,13 +8,29 @@ namespace Microsoft.Extensions.VectorData;
 /// Represents a generic data model that can be used to store and retrieve any data from a vector store.
 /// </summary>
 /// <typeparam name="TKey">The data type of the record key.</typeparam>
-/// <param name="key">The key of the record.</param>
-public sealed class VectorStoreGenericDataModel<TKey>(TKey key)
+public sealed class VectorStoreGenericDataModel<TKey>
 {
+    /// <summary>
+    /// Constructs a new <see cref="VectorStoreGenericDataModel{TKey}"/>.
+    /// </summary>
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring as nullable.
+    public VectorStoreGenericDataModel()
```

Comment:
@roji in the plans that you have for refactoring `VectorStoreGenericDataModel<TKey>` into `Dictionary<string, object>`, is there a room for making `TRecord : new()`?